plugins/org.python.pydev/src/org/python/pydev/editor/correctionassist/heuristics/AssistFString.java

```diff
@@ -0,0 +1,132 @@
+import org.python.pydev.shared_core.code_completion.IPyCompletionProposal;
+import org.python.pydev.shared_core.image.IImageCache;
+import org.python.pydev.shared_core.image.IImageHandle;
+import org.python.pydev.shared_core.image.UIConstants;
+import org.python.pydev.shared_core.partitioner.FastPartitioner;
+import org.python.pydev.shared_core.string.FastStringBuffer;
+
+public class AssistFString implements IAssistProps {
+    @Override
+    public List<ICompletionProposalHandle> getProps(PySelection ps, IImageCache imageCache, File f,
+            IPythonNature nature, IPyEdit edit, int offset) throws BadLocationException, MisconfigurationException {
+        PySourceViewer viewer = null;
+        if (edit instanceof PyEdit) { //only in tests it is actually null
+            viewer = ((PyEdit) edit).getPySourceViewer();
+        }
+        List<ICompletionProposalHandle> lst = new ArrayList<ICompletionProposalHandle>();
+
+        IDocument doc = ps.getDoc();
+        ITypedRegion partition = ((FastPartitioner) PyPartitionScanner.checkPartitionScanner(doc)).getPartition(offset);
+
+        FastStringBuffer strBuf = new FastStringBuffer();
+        strBuf.append('f').append(doc.get(partition.getOffset(), partition.getLength()));
+
+        FastStringBuffer variablesBuf = new FastStringBuffer();
+        // get string end offset
+        int partitionEndOffset = partition.getOffset() + partition.getLength();
+        int i = partitionEndOffset;
+        try {
+            // get format end offset
+            ParsingUtils parsingUtils = ParsingUtils.create(doc);
+            while (i < parsingUtils.len()) {
+                char c = parsingUtils.charAt(i);
+                if (Character.isWhitespace(c)) {
+                    i++;
+                    continue;
+                }
+                if (c == '{' || c == '[' || c == '(' || c == '"' || c == '\'') {
+                    i = parsingUtils.eatPar(i, null, c) + 1;
+                    break;
+                }
+                if (Character.isJavaIdentifierPart(c)) {
+                    i++;
+                    while (i < parsingUtils.len()) {
+                        c = parsingUtils.charAt(i);
+                        if (Character.isJavaIdentifierPart(c) || c == '.') {
+                        } else {
+                            if (c == '{' || c == '[' || c == '(' || c == '"' || c == '\'') {
+                                i = parsingUtils.eatPar(i, null, c);
+                            }
+                            break; // inner break
+                        }
+                        i++;
+                    }
+                    continue;
+                }
+                i++;
+            }
+
+            variablesBuf.append(doc.get(partitionEndOffset, i - partitionEndOffset));
+            // get only the variable names and remove the last comma
+            variablesBuf.trim().deleteFirst();
+            if (variablesBuf.trim().startsWith('(')) {
```

Comment:
Shouldn't this check for other chars too? (`[`, `'`, etc.)?

It'd be nice to have tests for these use-cases too.